src/ufoLib2/objects/font.py

```diff
@@ -202,56 +204,72 @@ def read(cls, reader, lazy=True):
+        if self.layers.defaultLayer is not None:
+            del self.layers.defaultLayer[name]
+        raise Error(f"No default layer present, cannot delete '{name}'.")
+
+    def __getitem__(self, name: str) -> Glyph:
+        if self.layers.defaultLayer is not None:
+            return self.layers.defaultLayer[name]
+        raise Error(f"No default layer present, cannot retrieve '{name}'.")
+
+    def __setitem__(self, name: str, glyph: Glyph) -> None:
+        if self.layers.defaultLayer is not None:
+            self.layers.defaultLayer[name] = glyph
+        raise Error(f"No default layer present, cannot insert '{name}'.")
+
+    def __iter__(self) -> Iterator[Glyph]:
+        if self.layers.defaultLayer is not None:
+            return iter(self.layers.defaultLayer)
+        raise Error("No default layer present, cannot iterate over it.")
+
+    def __len__(self) -> int:
+        if self.layers.defaultLayer is not None:
+            return len(self.layers.defaultLayer)
+        raise Error("No default layer present, cannot determine glyph count.")
+
+    def get(self, name: str, default: Optional[T] = None) -> Union[Optional[T], Glyph]:
         """Return the :class:`.Glyph` object for name if it is present in the
         default layer, otherwise return ``default``."""
-        return self.layers.defaultLayer.get(name, default)
+        if self.layers.defaultLayer is not None:
+            return self.layers.defaultLayer.get(name, default)
+        return default
 
-    def keys(self):
+    def keys(self) -> KeysView[str]:
         """Return a list of glyph names in the default layer."""
-        return self.layers.defaultLayer.keys()
+        if self.layers.defaultLayer is not None:
+            return self.layers.defaultLayer.keys()
+        raise Error("No default layer present, cannot retrieve list of glyph names.")
 
-    def close(self):
+    def close(self) -> None:
         """Closes the UFOReader if it still exists to finalize any outstanding
         file operations."""
         if self._reader is not None:
             self._reader.close()
 
-    def __enter__(self):
+    def __enter__(self) -> "Font":
         # TODO: Document an example for this.
         return self
 
-    def __exit__(self, exc_type, exc_value, exc_tb):
+    def __exit__(self, exc_type: Any, exc_value: Any, exc_tb: Any) -> None:
         self.close()
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         names = list(filter(None, [self.info.familyName, self.info.styleName]))
         fontName = " '{}'".format(" ".join(names)) if names else ""
         return "<{}.{}{} at {}>".format(
             self.__class__.__module__, self.__class__.__name__, fontName, hex(id(self))
         )
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         # same as attrs-defined __eq__ method, only that it un-lazifies fonts if needed
-        if other.__class__ is not self.__class__:
+        if not isinstance(other, Font):
```

Comment:
here you replaced `self.__class__` with `Font` and it's not really the same because now this method will happily compare this Font with another object that is a subclass of Font but has extra members that need special care to be compared properly?